### DISCUSSION TITLE
Reduce report file size: Plotly -- include_plotlyjs="cdn"

### DISCUSSION THREAD

**Discussion by lukefullard** [datapane/datapane#171] [Ideas]
Hello all,
I was thinking about the size of some reports (tens of Mb for some of my files) which makes for slow loading times in the browser.

In plotly, when exporting a figure to html we have the option to exclude the background javascript in the file which takes a plot from ~3Mb to ~100 Kb.
Basically when you open the plotly html file the browser downloads the javascript from the plotly server (I think).
This makes sharing a 100Kb file easy, rather than a 3Mb file.

Is something like this possible with Datapane?
Thanks

[Ref1](https://plotly.com/python/interactive-html-export/)
[Ref2](https://stackoverflow.com/questions/66368233/what-does-include-plotlyjs-cdn-in-plotly-python-do-to-the-generated-html)

**Comment by mands** [comment]
Hi @lukefullard 

Yes we've investigated that in the past, but it was made a bit more difficult to handle different versions of the plotly (and other plotting libraries). Plus many have mentioned preferring the ability of having the fully embedded file that doesn't require any network access to view. However we'll add this to our product board to think about further.

Have you tried using the public server at datapane.com ?
